lib/elixir/lib/record.ex

```diff
@@ -309,7 +311,7 @@ defmodule Record do
 
   # Callback invoked from record/0 and record/1 macros.
   @doc false
-  def __access__(atom, fields, args, caller) do
+  def __access__(atom, fields, args, caller) when is_atom(atom) and is_map(caller) do
```

Comment:
True. I tested it and it will raise a `FunctionClauseError` on the `__access__` functions, which won't help at all.
On the other hand, it is nowhere checked for example the tag value to be an atom, and I can use a string for example, or any other term that when quoted remains the same, as strings or lists.


```elixir
iex(1)> defmodule User do
...(1)>   require Record
...(1)>   Record.defrecord :user, "user", [name: "meg", age: "25"]
...(1)> end
iex(2)> import User
iex(3)> user()
{"user", "meg", "25"}
```